ironfish/src/primitives/transaction.ts

```diff
@@ -189,7 +192,7 @@ export class Transaction {
    * Used for cases where a signature needs to be commited to in the hash like P2P transaction gossip
    */
   hash(): TransactionHash {
-    return blake3(this.transactionPostedSerialized)
+    return this._hash
```

Comment:
Yea only reason I reverted it is because we have a test that does comparison of transactions before and after the hash method is called. It was kind of deeply nested: we're comparing block equality in that test and the transaction is just one of the transactions in the block. It was failing because _hash was different after the hash method was called. So it would require some effort to rewrite. 

I don't like making source code changes just for tests but in this case I figured since blake3 is fast it would be somewhat negligible to the other logic we're doing in the constructor and it would save some time in having to refactor the test. I can spend that time upfront but I think it would be better to mark this as a TODO given the speed of blake3. Wdyt?

Reply:
Chatted about this on Slack, ended up going with memoizing the hash and hacking in a fix to the test.